optuna/visualization/matplotlib/_hypervolume_history.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+from typing import NamedTuple
+from typing import Sequence
+
+import numpy as np
+
+from optuna._experimental import experimental_func
+from optuna._hypervolume import WFG
+from optuna.logging import get_logger
+from optuna.samplers._base import _CONSTRAINTS_KEY
+from optuna.study import Study
+from optuna.study._multi_objective import _get_pareto_front_trials_by_trials
+from optuna.study._study_direction import StudyDirection
+from optuna.trial import TrialState
+from optuna.visualization.matplotlib._matplotlib_imports import _imports
+
+
+if _imports.is_successful():
+    from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
+
+_logger = get_logger(__name__)
+
+
+class _HypervolumeHistoryInfo(NamedTuple):
+    trial_numbers: list[int]
+    values: list[float]
+
+
+@experimental_func("3.3.0")
+def plot_hypervolume_history(
+    study: Study,
+    reference_point: Sequence[float],
+) -> "Axes":
+    """Plot hypervolume history of all trials in a study with Matplotlib.
+
+    Example:
+
+        The following code snippet shows how to plot optimization history.
+
+        .. plot::
+
+            import optuna
+            import matplotlib.pyplot as plt
+
+
+            def objective(trial):
+                x = trial.suggest_float("x", 0, 5)
+                y = trial.suggest_float("y", 0, 3)
+
+                v0 = 4 * x ** 2 + 4 * y ** 2
+                v1 = (x - 5) ** 2 + (y - 5) ** 2
+                return v0, v1
+
+
+            study = optuna.create_study(directions=["minimize", "minimize"])
+            study.optimize(objective, n_trials=50)
+
+            reference_point=[100, 50]
+            optuna.visualization.matplotlib.plot_hypervolume_history(study, reference_point)
+            plt.tight_layout()
+
+        .. note::
+            You need to adjust the size of the plot by yourself using ``plt.tight_layout()`` or
+            ``plt.savefig(IMAGE_NAME, bbox_inches='tight')``.
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their hypervolumes.
+           ``study.n_objectives`` must be 2 or more.
+
+        reference_point:
+            A reference point to use for hypervolume computation.
+
+    Returns:
+        A :class:`matplotlib.axes.Axes` object.
+    """
+
+    _imports.check()
+
+    assert study._is_multi_objective(), (
```

Comment:
It's better to raise ValueError for this, I think.

Reply:
done